src/Microsoft.IdentityModel.Tokens/CryptoProviderCacheOptions.cs

```diff
@@ -37,28 +37,80 @@ namespace Microsoft.IdentityModel.Tokens
+        /// <summary>
+        /// Gets or sets the percentage of the cache to be removed when _maxCapacityPercentage is reached.
+        /// </summary>
+        public double CompactionPercentage
+        {
+            get => _compactionPercentage;
+            set => _compactionPercentage = (value >= MinCompactionPercentageValue && value <= MaxCompactionPercentageValue) ? value : throw LogHelper.LogExceptionMessage(new ArgumentOutOfRangeException(nameof(CompactionPercentage), LogHelper.FormatInvariant(LogMessages.IDX10903, MinCompactionPercentageValue, MaxCompactionPercentageValue, value)));
+        }
+
+        /// <summary>
+        /// Default percentage of _capacity, when reached, _compactionPercentage% of the cache will be removed.
+        /// </summary>
+        public const double DefaultMaxCapacityPercentage = .95;
+
+        private const double MinCapacityPercentageValue = 0.5;
+        private const double MaxCapacityPercentageValue = 1.0;
+        private double _maxCapacityPercentage = DefaultMaxCapacityPercentage;
+        /// <summary>
+        /// Gets or sets the percentage of _capacity, when reached, _compactionPercentage% of the cache will be removed.
+        /// </summary>
+        public double MaxCapacityPercentage
+        {
+            get => _maxCapacityPercentage;
+            set => _maxCapacityPercentage = (value >= MinCapacityPercentageValue && value <= MaxCapacityPercentageValue) ? value : throw LogHelper.LogExceptionMessage(new ArgumentOutOfRangeException(nameof(MaxCapacityPercentage), LogHelper.FormatInvariant(LogMessages.IDX10904, MinCapacityPercentageValue, MaxCapacityPercentageValue, value)));
+        }
+
+        /// <summary>
+        /// The default value of the period to wait to remove expired items, in seconds.
+        /// </summary>
+        public const int DefaultRemoveExpiredValuesIntervalInSeconds = 300;
+
+        private int _removeExpiredValuesIntervalInSeconds = DefaultRemoveExpiredValuesIntervalInSeconds;
+        /// <summary>
+        /// Gets or sets the period to wait to remove expired items, in seconds.
+        /// </summary>
+        public int RemoveExpiredValuesIntervalInSeconds
+        {
+            get => _removeExpiredValuesIntervalInSeconds;
+            set => _removeExpiredValuesIntervalInSeconds = (value > 0) ? value : throw LogHelper.LogExceptionMessage(new ArgumentOutOfRangeException(nameof(RemoveExpiredValuesIntervalInSeconds), LogHelper.FormatInvariant(LogMessages.IDX10905, value)));
+        }
+
+        /// <summary>
+        /// Gets or sets the value that determines whether or not to remove expired values.
+        /// </summary>
+        public bool RemoveExpiredValues { get; set; } = false;
+
+        /// <summary>
+        /// The desired cache type for caching providers. Defaults to LRU.
+        /// </summary>
+        public ProviderCacheType CacheType { get; set; } = ProviderCacheType.LRU;
```

Comment:
The CryptoProviderCache most likely will never use LRU, do we need this public?

Reply:
My idea is to allow users to select the cache types based on the needs of applications. For example, the IC3 team who has expressed major concerns on our LRU using tasks while they have only a few signature providers in the cache. It provides an option for them to just use the new MaximumSizeCache or any new cache types we may have in the future.